layers/ipfix.go

```diff
@@ -0,0 +1,773 @@
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 3 - Field Value 1    |   Record 3 - Field Value 2    |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 3 - Field Value 3    |             ...               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |              ...              |      Padding (optional)       |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+type IPFIXDataRecord struct {
+	// Template or Option template ID use to encode the record
+	ID IPFIXSetIDType
+	// List of fields in that template record.
+	// Order matter as it respect the same order of Template Field Specifiers.
+	Fields []IPFIXField
+}
+
+func (dr *IPFIXDataRecord) decodeFromBytes(data *[]byte, t IPFIXTemplateAccessor) (int, error) {
+	dataRead := 0
+	if t == nil {
+		return dataRead, fmt.Errorf("cannot decode data record without a template record")
+	}
+
+	if dr.ID != t.GetID() {
+		return dataRead, fmt.Errorf("invalid template ID")
+	}
+
+	if err := t.IterFieldSpecifier(func(fs *IPFIXFieldSpecifier) error {
+		f := &IPFIXField{
+			InformationElementID: fs.InformationElementID,
+			EnterpriseNumber:     fs.EnterpriseNumber,
+		}
+		n, err := f.decodeFromBytes(data, fs)
+		if err != nil {
+			return err
+		}
+		dr.Fields = append(dr.Fields, *f)
+		dataRead += n
+
+		return nil
+	}); err != nil {
+		return dataRead, err
+	}
+
+	return dataRead, nil
+}
+
+// IPFIXField is a Field in a Data Recors (RFC 7011 section 3.4.3)
+//
+//      0                   1                   2                   3
+//      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Set ID = Template ID        |          Length               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 1 - Field Value 1    |   Record 1 - Field Value 2    |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 1 - Field Value 3    |             ...               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 2 - Field Value 1    |   Record 2 - Field Value 2    |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 2 - Field Value 3    |             ...               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 3 - Field Value 1    |   Record 3 - Field Value 2    |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |   Record 3 - Field Value 3    |             ...               |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//     |              ...              |      Padding (optional)       |
+//     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Just depicting generic field value with fixed length and varying length is good?